fix: Allow deprecated features to be removed between k8s versions

michaelasp opened the pull request:
If a feature gate is marked as deprecated between versions then it is reasonable for the next version to have the feature removed. Add this to tests to fix failures.

Fixes #35179

michaelasp commented:
/cc @Jefftree 
/cc @BenTheElder

Jefftree commented:
/ok-to-test

BenTheElder commented on experiment/compatibility-versions/validate-compatibility-versions-feature-gates.sh:
```diff
@@ -168,6 +171,11 @@ for feature_name in "${!expected_stage[@]}"; do
     fi
 
+    # Deprecated feature that was removed, these can continue.
+    if [[ echo "$DEPRECATED_FEATURE_LIST" | tr '|' '\n' | grep -q -w "$feature_name" ]]; then
```
```suggestion
    if [[  "$DEPRECATED_FEATURE_LIST" == *"$feature_name"* ]]; then
```

michaelasp commented:
> We might want to specify that this is intended only for featuregates read exclusively by components that are out of scope for emulation version? (so kubelet in this case).
> 
> Maybe update the PR title and code comments etc.
> 
> We don't want someone finding this and adding an api-server feature that should've been lockedtodefault instead of removed.

Yeah makes sense, added comments everywhere that someone might see the use of this and updated pr title/description.

BenTheElder reviewed (commented):
/lgtm
/approve
